website/source/docs/guides/connect-gateways.md

```diff
@@ -0,0 +1,421 @@
+stanza in its registration file. For this guide, you can use socat to act as a
+backend service and register a dummy service called web to represent the client
+service. Those names are used in our examples. If you have services that you
+would like to connect, feel free to use those instead.
+
+~> **Caution:** Connect takes its default intention policy from Consul’s default
+ACL policy. If you have set your default ACL policy to deny (as is recommended
+for secure operation) and are adding Connect to already registered services,
+those services may lose connection to each other until you set an intention
+between them to allow communication.
+
+### Register a back end service in one datacenter
+
+In one datacenter register a backend service and add an Envoy sidecar proxy
+registration. To do this you will either create a new registration file or edit
+an existing one to include a sidecar proxy stanza. If you are using socat as
+your backend service, you will create a new file called `socat.json` that will
+contain the below snippet. Since you have ACLs enabled, you will have to [create
+a token for the service](/consul/security-networking/production-acls#apply-individual-tokens-to-the-services).
+
+```json
+{
+  "service": {
+    "name": "socat",
+    "port": 8181,
+    "token": "<token here>",
+    "connect": {"sidecar_service": {} }
+  }
+}
+```
+
+Note the Connect stanza of the registration with the `sidecar_service` and
+`token` options. This is what you would add to an existing service registration
+if you are not using socat as an example.
+
+Reload the client with the new or modified registration.
+
+```text
+$ consul reload
+```
+
+Then start Envoy specifying which service it will proxy.
+
+```text
+$ consul connect envoy -sidecar-for socat
+```
+
+If you are using socat as your example, start it now on the port you specified
+in your registration by running the following command.
+
+```text
+$ socat -v tcp-l:8181,fork exec:"/bin/cat"
+```
+
+Check that the socat service is running by accessing it using netcat on the same
+node. It will echo back anything you type.
+
+```text
+$ nc 127.0.0.1 8181
+hello
+hello
+echo
+echo
+```
+
+Stop the running netcat service by typing `ctrl + c`.
+
+### Register a front end service in the other datacenter
+
+Now in your other datacenter, you will register a service (with a sidecar proxy)
+that calls your backend service. Your registration will need to list the backend
+service as your upstream. Like the backend service, you can use an example
+service, which will be called web, or append the connect stanza to an existing
+registration with some customization.
+
+To use web as your front end service, create a registration file called
+`web.json` that contains the following snippet.
+
+```json
+{
+  "service": {
+    "name": "web",
+    "port": 8080,
+    "token": "<token here>",
+    "connect": {
+      "sidecar_service": {
+        "proxy": {
```

Comment:
Per my other comment above, if you don't enable central configuration you can set up gateways per proxy/upstream like:

Per Proxy
```json
{ 
   "proxy": {
      "mesh_gateway": {
         "mode": "local"
      }
      ...
   }
}
```

Per Upstream
```json
{ 
   "proxy": {
      "upstreams": [
         {
            "mesh_gateway": {
               "mode": "local"
            },
            ...
         },
         ...
      ]
      ...
   }
}